include/core/Discretization.hpp

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include "span.hpp"
+
+using GeoDivide = int;
+
+struct Discretization
+{
+    virtual ~Discretization() = default;
+
+    // Aligned with the possible results from BMI's get_grid_type
+    enum class GridType {
+        scalar,
+        points,
+        vector,
+        unstructured,
+        structured_quadrilateral,
+        rectilinear,
+        uniform_rectilinear
+    };
+    virtual GridType grid_type() const = 0;
+
+    // Must cover an identified geographic domain
+    virtual GeoDivide divide() const = 0;
```

Comment:
Are discretizations only associated to catchments, and are those catchment's divide realizations the return of this member?

_if not_ (and this is mostly semantics), I wouldn't call this a `divide`, since that term is specific to catchment boundaries. If the intention of this member is to retrieve the geographic domain of discretizations over arbitrary geographic domains, this should be probably be called `extent` and return something like a `GeoExtent`.

Reply:
Good catch on the semantics. I think `extent` is right, since I intend it to be possible for a `Discretization` to cover the entirety of a gridded model's operational domain, spanning distinct catchments.